server/routes/login_routes/welcome.js

```diff
@@ -0,0 +1,51 @@
+const querystring = require('querystring');
+const superagent = require('superagent');
+
+
+const payload = facebookCode =>
+  querystring.stringify({
+    code: facebookCode,
+    client_id: process.env.FB_CLIENT,
+    client_secret: process.env.FB_SECRET,
+    redirect_uri: `${process.env.BASE_URL}/welcome`,
+  });
+
+
+const getAccessToken = (code, callback) => {
+  const qs = payload(code);
+  const url = `https://graph.facebook.com/v2.3/oauth/access_token?${qs}`;
+  superagent
+    .get(url)
+    .set('Accept', 'application/json')
+    .end((err, response) => {
+      if (err) callback(err);
+      else callback(err, response.body.access_token);
+    });
+};
+
+const getUserDetails = (accessToken, callback) => {
+  const userDetailsUrl = `https://graph.facebook.com/me?access_token=${accessToken}`;
+  superagent
+    .get(userDetailsUrl)
+    .set('Accept', 'application/json')
+    .end((err, response) => {
+      if (err) callback(err);
+      callback(null, response.body);
+    });
+};
+
+const welcome = {
+  method: 'GET',
+  path: '/welcome',
+  handler(request, reply) {
+    getAccessToken(request.query.code, (accessTokenError, accessToken) => {
+      if (accessTokenError) throw accessTokenError;
```

Comment:
yes, maybe we should raise an  401 or 403, we were talking about handle it differently, but ad the end we left it like that.